src/interface/control_textbox.cpp

```diff
@@ -361,102 +574,125 @@ ControlTextbox::get_string() const
 }
 
 std::string
-ControlTextbox::get_contents() const
+ControlTextbox::get_contents(int line) const
```

Comment:
This does handle negative logic too? I think it can be made clear with a comment.